src/tools/rest-backend/model_user.hpp

```diff
@@ -0,0 +1,355 @@
+/**
+ * @file
+ *
+ * @brief model for user
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#ifndef ELEKTRA_REST_MODEL_USER_HPP
+#define ELEKTRA_REST_MODEL_USER_HPP
+
+#include <map>
+#include <string>
+#include <utility>
+
+#include <boost/algorithm/string/compare.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+
+#include <config.hpp>
+#include <exceptions.hpp>
+#include <kdb_includes.hpp>
+
+/**
+ * @brief main namespace for the REST service
+ */
+namespace kdbrest
+{
+
+/**
+ * @brief namespace for models
+ */
+namespace model
+{
+
+/**
+ * @brief model class for a REST service user
+ * 
+ * this class encapsulates all information that belongs to a user.
+ */
+class User : public kdb::Key
+{
+
+public:
+	/**
+     * @brief constructs an User object based on a kdb::Key object
+	 * 
+     * @param key A kdb::Key object
+     */
+	inline User (kdb::Key & k) : kdb::Key (k)
```

Comment:
useless inline (constructor/methods are implicitly inline when directly implemented in the class)